app/src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.java

```diff
@@ -1075,8 +1080,9 @@ protected void onResume() {
         syncAndUpdateFolder(false, true);
 
         OCFile startFile = null;
-        if (getIntent() != null && getIntent().getParcelableExtra(EXTRA_FILE) != null) {
-            startFile = getIntent().getParcelableExtra(EXTRA_FILE);
+        OCFile fileArgs = IntentExtensionsKt.getParcelableArgument(getIntent(), EXTRA_FILE, OCFile.class);
+        if (getIntent() != null && fileArgs != null) {
+            startFile = fileArgs;
```

Comment:
```suggestion
        if (getIntent() != null) {
            OCFile fileArgs = IntentExtensionsKt.getParcelableArgument(getIntent(), EXTRA_FILE, OCFile.class);
            if (fileArgs != null) {
                startFile = fileArgs;
                setFile(startFile);
            }
```

The original code checked whether `getIntent()` is `null` before making any further calls. Apparently there are certain edge cases where this could occur. In such cases, `getParcelableArgument` is called on a null-object. I'm not sure, whether it's worth considering such cases.